app/src/main/java/com/swent/assos/ui/screens/calendar/Calendar.kt

```diff
@@ -1,80 +1,216 @@
-          dayWidth = dayWidth,
-          hourHeight = hourHeight,
-          data = data,
-          modifier =
-              Modifier.weight(1f)
-                  .verticalScroll(verticalScrollState)
-                  .horizontalScroll(horizontalScrollState))
+  val selectedEvents = calendarViewModel.selectedEvents.collectAsState()
+  val selectedDate = calendarViewModel.selectedDate.collectAsState()
+
+  val verticalScrollState = rememberScrollState(1350)
+
+  LaunchedEffect(key1 = Unit) { calendarViewModel.updateEvents() }
+
+  LaunchedEffect(events.value, selectedDate.value) { calendarViewModel.filterEvents() }
+
+  Scaffold(
+      modifier = Modifier.semantics { testTagsAsResourceId = true }.testTag("CalendarScreen"),
+      topBar = { PageTitle(title = "Calendar - ${selectedDate.value.format(dateFormatter)}") }) {
+        Column(modifier = Modifier.padding(16.dp).padding(it).fillMaxSize()) {
+          InfiniteScrollableDaysList(
+              selectedDate = selectedDate,
+              onDateSelected = { newDate -> calendarViewModel.updateSelectedDate(newDate) })
+
+          Spacer(modifier = Modifier.height(16.dp))
+          Text(
+              text =
+                  if (selectedDate.value == LocalDate.now()) {
+                    "Schedule Today"
+                  } else {
+                    "Daily Schedule"
+                  },
+              fontSize = 16.sp,
+              fontWeight = FontWeight.SemiBold,
+              color = Color(0xFF1E293B))
+
+          Spacer(modifier = Modifier.height(32.dp))
+          DailySchedule(
+              events = selectedEvents,
+              verticalScrollState = verticalScrollState,
+              eventContent = eventContent)
+          Spacer(modifier = Modifier.height(32.dp))
+
+          Reminder(calendarViewModel = calendarViewModel)
+        }
+      }
+}
+
+@Composable
+fun InfiniteScrollableDaysList(
+    selectedDate: State<LocalDate>,
+    onDateSelected: (LocalDate) -> Unit
+) {
+  val startDate = LocalDate.now().minusDays(3L) // Start from 3 days ago
+
+  // Generate a list of LocalDate objects representing days
+  val daysList = remember {
+    List(365 * 10) { // Generating 10 years worth of days
+      startDate.plusDays(it.toLong())
     }
   }
+
+  LazyRow(modifier = Modifier.testTag("DaysList")) {
+    // Loop through the list of days infinitely
+    itemsIndexed(daysList) { index, day ->
+      DayItem(date = day, selected = day == selectedDate.value, onDateSelected)
+      if (day == selectedDate.value ||
+          index < daysList.size - 1 && daysList[index + 1] == selectedDate.value) {
+        // Adjust spacing between the selected data
+        Spacer(modifier = Modifier.width(14.5.dp))
+      } else {
+        // Adjust spacing between other dates
+        Spacer(modifier = Modifier.width(26.dp))
+      }
+    }
+  }
+}
+
+@Composable
+fun DayItem(date: LocalDate, selected: Boolean, onDateSelected: (LocalDate) -> Unit) {
```

Comment:
Sonarcloud spots a high maintainability issue that is high cognitive complexity. 
There are lots of if else for the same argument, maybe it is possible to reduce the number of if's